From 37b71a57b06ebb9032b9620a9b358bb64f868b51 Mon Sep 17 00:00:00 2001
From: George Chang <georgekgchang@google.com>
Date: Wed, 5 Jun 2019 15:32:46 +0800
Subject: [PATCH 3/3] Prevent OOB Read in Mfc_Transceive

Bug: 125900276
Test: Manual test/ Mifare Tag Read/Write
Change-Id: Id142d52ef682db4d4605d0e8e3d0516dacfd8d8a
(cherry picked from commit a5430db011f8005799d4c48b4fe6c7691733ea7c)
---
 nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c b/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
index 801209f4..50e2deea 100755
--- a/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
+++ b/nci/jni/extns/pn54x/src/mifare/phNxpExtns_MifareStd.c
@@ -1081,6 +1081,11 @@ NFCSTATUS Mfc_Transceive(uint8_t *p_data, uint32_t len)
     EXTNS_SetCallBackFlag(TRUE);
     if( p_data[0] == 0x60 || p_data[0] == 0x61 )
     {
+        if ( len < 12 )
+        {
+            android_errorWriteLog(0x534e4554, "125900276");
+            return status;
+        }
 
         NdefMap->Cmd.MfCmd = p_data[0];
 
-- 
2.17.1

